### DISCUSSION TITLE
New to Everything - please be kind

### DISCUSSION THREAD

**Discussion by MaestroMcCauley** [airsdk/Adobe-Runtime-Support#2374] [Q&A]
My copy of Sound Set Editor has no User Guide - when I select the.  This Avid product uses Air_Runtime.  Is this an Avid problem or a Harmon problem?   
![Screen Shot 2022-12-21 at 10 18 13 AM](https://user-images.githubusercontent.com/121182215/208952904-d46c9b99-d46e-486b-b629-a4fa0c1e72b6.png)
I am very new to git-hub, and using the Terminal. Searching the discussions for this there was nothing.  Perhaps I need guidance on how to search??

**Comment by ajwfrost** [comment]
Hi

Welcome :-) 
Having had a quick search for "Sound Set Editor", it looks like it's an application where they've asked you to install AIR first (their website links to Adobe's original page which should - via a few steps - have meant you ended up at airsdk.harman.com/runtime

Did you then download and install the AIR runtime using the "Adobe AIR for MacOS" link at the top of that page? Given the blank screen on the user guide, I am wondering whether they were using some HTML-based (or PDF-based) code for this, and the latest AIR runtime doesn't support that in quite the same way...

Can I ask you to please try:
- using Finder, go to the "Applications" and find the Adobe AIR application installer: if you double-click on this, it should then allow you to uninstall the AIR runtime again.
- then go back to our website and download the lower version, 33.1.1.744 - direct link is https://airsdk.harman.com/assets/downloads/33.1.1.744/AdobeAIR.dmg
- once you've installed that, please re-run the Sound Set Editor (no need to uninstall or re-install it)

Hopefully that should work...

thanks

**Comment by MaestroMcCauley** [comment]
Doing what you've instructed I get this message:

I uninstalled, and clicked on the link - what did I do incorrectly?  I installed AIR first.   Your help is appreciated.
Bob

    On Wednesday, December 21, 2022 at 10:44:19 AM CST, Andrew Frost ***@***.***> wrote:  
 
 


Hi

Welcome :-)
Having had a quick search for "Sound Set Editor", it looks like it's an application where they've asked you to install AIR first (their website links to Adobe's original page which should - via a few steps - have meant you ended up at airsdk.harman.com/runtime

Did you then download and install the AIR runtime using the "Adobe AIR for MacOS" link at the top of that page? Given the blank screen on the user guide, I am wondering whether they were using some HTML-based (or PDF-based) code for this, and the latest AIR runtime doesn't support that in quite the same way...

Can I ask you to please try:
   
   - using Finder, go to the "Applications" and find the Adobe AIR application installer: if you double-click on this, it should then allow you to uninstall the AIR runtime again.
   - then go back to our website and download the lower version, 33.1.1.744 - direct link is https://airsdk.harman.com/assets/downloads/33.1.1.744/AdobeAIR.dmg
   - once you've installed that, please re-run the Sound Set Editor (no need to uninstall or re-install it)

Hopefully that should work...

thanks

**Comment by MaestroMcCauley** [comment]
Now when I try to open the User Guide in the Sound Set Editor, the software crashes - this uses the lower version of Air that I downloaded from your link below.  I am at a loss...
Bob

    On Wednesday, December 21, 2022 at 10:44:19 AM CST, Andrew Frost ***@***.***> wrote:  
 
 


Hi

Welcome :-)
Having had a quick search for "Sound Set Editor", it looks like it's an application where they've asked you to install AIR first (their website links to Adobe's original page which should - via a few steps - have meant you ended up at airsdk.harman.com/runtime

Did you then download and install the AIR runtime using the "Adobe AIR for MacOS" link at the top of that page? Given the blank screen on the user guide, I am wondering whether they were using some HTML-based (or PDF-based) code for this, and the latest AIR runtime doesn't support that in quite the same way...

Can I ask you to please try:
   
   - using Finder, go to the "Applications" and find the Adobe AIR application installer: if you double-click on this, it should then allow you to uninstall the AIR runtime again.
   - then go back to our website and download the lower version, 33.1.1.744 - direct link is https://airsdk.harman.com/assets/downloads/33.1.1.744/AdobeAIR.dmg
   - once you've installed that, please re-run the Sound Set Editor (no need to uninstall or re-install it)

Hopefully that should work...

thanks

**Reply by ajwfrost** [reply]
Hi

That sounds good (well, at least it probably confirms what I was thinking about how they're doing that user manual). If it crashes, that implies you're running on Apple/ARM hardware... the 33.1.1.744 build contains the webkit component that Adobe had created many years ago, and this is the piece that's missing from the latter runtimes. But this component is only available for the Intel platform so it's not possible to load it into a process running as an ARM executable.

The solution then is to force your application to run in Intel mode, which you can do by right-clicking on the app in Finder, choosing "Get Info", and then check the box that says "Open using Rosetta". Once this is ticked, it will automatically open the application in Intel mode and the webkit library can then be loaded into it when you view that user guide.

Hope that helps!

**Comment by MaestroMcCauley** [comment]
I'm feeling useless.

I downloaded Air_SDK yesterday.  I cannot locate it to do what you instruct below.  See the screen grab below.  Am I blind?  Where would it be?
Bob





    On Thursday, December 22, 2022 at 12:21:43 AM CST, Andrew Frost ***@***.***> wrote:  
 
 


Hi

That sounds good (well, at least it probably confirms what I was thinking about how they're doing that user manual). If it crashes, that implies you're running on Apple/ARM hardware... the 33.1.1.744 build contains the webkit component that Adobe had created many years ago, and this is the piece that's missing from the latter runtimes. But this component is only available for the Intel platform so it's not possible to load it into a process running as an ARM executable.

The solution then is to force your application to run in Intel mode, which you can do by right-clicking on the app in Finder, choosing "Get Info", and then check the box that says "Open using Rosetta". Once this is ticked, it will automatically open the application in Intel mode and the webkit library can then be loaded into it when you view that user guide.

Hope that helps!

**Reply by ajwfrost** [reply]
Hi

The screen grab isn't showing up for me here ..? but it's not the SDK that you would need. To work around the issue, with what I think you have currently set up/installed, you would need to open Finder and then go to Applications, then find the "Sound Set Editor" application and right-click on this. It's this one that needs to be changed so that it runs using Rosetta...

thanks

**Comment by MaestroMcCauley** [comment]
Thanks for all your help, however the SSE has no box to click or mention of Rosetta when I right click.  I would do a screen grab, but you aren't seeing them.
Thank you again,Bob

    On Thursday, December 22, 2022 at 07:14:10 AM CST, Andrew Frost ***@***.***> wrote:  
 
 


Hi

The screen grab isn't showing up for me here ..? but it's not the SDK that you would need. To work around the issue, with what I think you have currently set up/installed, you would need to open Finder and then go to Applications, then find the "Sound Set Editor" application and right-click on this. It's this one that needs to be changed so that it runs using Rosetta...

thanks

**Reply by ajwfrost** [reply]
I think if you use the Github website rather than going through emails, it should be possible to do screenshots (or at least, it can cope with pasting in images..)

So right-click on SSE in Finder and you should see "Get Info" which then gives you a thin information panel about the application and its permissions etc. A little way down this it tells you if it's an Intel app or a Universal one, if it's Universal then it should have an option box to tick for using Rosetta..?

thanks

**Reply by MaestroMcCauley** [reply]
![Screen Shot 2022-12-22 at 5 48 33 PM](https://user-images.githubusercontent.com/121182215/209244163-111191e0-a958-449a-af23-9fc43a83d702.png)

This is what I see when I right-click SSE in Finder.

**Reply by ajwfrost** [reply]
Hmm... so I see:
![image](https://user-images.githubusercontent.com/6400804/209244855-c3636d80-201a-4cff-85cd-3dc09b8e12f4.png)

But this then implies that you're actually running already on Intel hardware? Maybe it was an assumption, can you please confirm what device/hardware you're actually running on ("About this mac" maybe shows the chip information)?

thanks

**Comment by MaestroMcCauley** [comment]
This can  wait.   There's a severe freeze coming.   Bob   thanks.

Sent from Yahoo Mail on Android 
 
  On Thu, Dec 22, 2022 at 5:58 PM, Andrew ***@***.***> wrote:   


Hmm... so I see:


But this then implies that you're actually running already on Intel hardware? Maybe it was an assumption, can you please confirm what device/hardware you're actually running on ("About this mac" maybe shows the chip information)?

thanks

**Reply by ajwfrost** [reply]
So I heard! I hope it's not too bad ...

Just looking at this one again, I've tried it on a M1 device and when opening the user guide, it just shows a white screen - just like your original post - and I'm using the build without the webkit library (i.e. the latest one).
I've just gone back to the 33.1.1.744 build and had to uninstall/reinstall the Sound Set Editor package. And then for some reason I can only open it when I've got Rosetta selected, but I still get the white screen where the PDF should be...

To actually get at the user guide, you can always right-click on the application and choose "Show package contents", then in Contents -> Resources you'll see the user guide PDF file that you can double-click on.

thanks

**Reply by MaestroMcCauley** [reply]
Your last paragraph actually WORKED!!!  Now I've forgotten what I needed the guide for.  I hope it helps in the long haul of setting up a Sound Set for IRCAM Solo Instruments!!

Thank You So Much!
Bob

**Reply by MaestroMcCauley** [reply]
I will read it!  Thank you again Andrew.

**Reply by ajwfrost** [reply]
Great, glad that helps. I'm not 100% sure why the actual display is showing up white .. it's an old application and I can't see what they're actually using to try to display the PDF, but we have seen other situations where people were relying on platform-specific libraries to view PDFs but then those plug-ins may not be present in the first place.. the AIR runtime that we ended up with does have an old version of WebKit in it, but it doesn't have a PDF viewer, so it would need to load something from the operating system, and it's highly likely that Apple have changed how that works or Adobe have changed what types of PDF plug-ins are available, etc....

Thanks and hope the icy weather doesn't last too long..
